plugins/source/aws/resources/services/iam/account_authorization_details.go

```diff
@@ -0,0 +1,93 @@
+package iam
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/apache/arrow/go/v14/arrow"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/aws/aws-sdk-go-v2/service/iam/types"
+	"github.com/cloudquery/cloudquery/plugins/source/aws/client"
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+	"github.com/cloudquery/plugin-sdk/v4/transformers"
+	sdkTypes "github.com/cloudquery/plugin-sdk/v4/types"
+)
+
+func AccountAuthorizationDetails() *schema.Table {
+	tableName := "aws_iam_account_authorization_details"
+	return &schema.Table{
+		Name:        tableName,
+		Description: `https://docs.aws.amazon.com/IAM/latest/APIReference/API_ManagedPolicyDetail.html`,
+		Resolver:    fetchIamAccountAuthorizationDetails,
+		Transform:   transformers.TransformWithStruct(&types.ManagedPolicyDetail{}),
+		Multiplex:   client.ServiceAccountRegionMultiplexer(tableName, "iam"),
+		Columns: []schema.Column{
+			client.DefaultAccountIDColumn(true),
+			{
+				Name:       "id",
+				Type:       arrow.BinaryTypes.String,
+				Resolver:   schema.PathResolver("PolicyId"),
+				PrimaryKey: true,
+			},
+			{
+				Name:     "tags",
+				Type:     sdkTypes.ExtensionTypes.JSON,
+				Resolver: resolveIamAccountAuthorizationDetailsTags,
+			},
+			{
+				Name:     "policy_version_list",
+				Type:     sdkTypes.ExtensionTypes.JSON,
+				Resolver: resolveIamPolicyVersionList,
+			},
+		},
+	}
+}
+
+func fetchIamAccountAuthorizationDetails(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
+	config := iam.GetAccountAuthorizationDetailsInput{
+		Filter: []types.EntityType{
+			types.EntityTypeAWSManagedPolicy, types.EntityTypeLocalManagedPolicy,
+		},
+	}
+	cl := meta.(*client.Client)
+	svc := cl.Services(client.AWSServiceIam).Iam
+	paginator := iam.NewGetAccountAuthorizationDetailsPaginator(svc, &config)
+
+	for paginator.HasMorePages() {
+		page, err := paginator.NextPage(ctx, func(options *iam.Options) {
+			options.Region = cl.Region
+		})
+		if err != nil {
+			return err
+		}
+		res <- page.Policies
```

Comment:
I have copied current codes of `iam_policies.go` on main branch into this file. As long as the API endpoint will never changes return values, this code should save current implementation.

Reply:
But because this is an entirely new table, users will have to make a change to begin using it.

Reply:
Sorry, I haven't understood your concern. To solve that, what can I do?

For example:
- make a migration script for users or an automated migration tool
- restore `iam_policies` table not to replace current table definition and name my implementation `aws_iam_policy_names` e.g.
- or other good idea

There are some idea for solution. What is suitable for you?

Reply:
So right now the table is only storing data from the `page.Policies` attribute, which is a `ManagedPolicyDetail`. But if we look at the [response object](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/iam#GetAccountAuthorizationDetailsOutput) it has many more fields. If a user syncs that table, we should be storing all of the data that is retrieved not just a single field

```
type GetAccountAuthorizationDetailsOutput struct {

	// A list containing information about IAM groups.
	GroupDetailList [][types](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/iam@v1.22.5/types).[GroupDetail](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/iam@v1.22.5/types#GroupDetail)

	// A flag that indicates whether there are more items to return. If your results
	// were truncated, you can make a subsequent pagination request using the Marker
	// request parameter to retrieve more items. Note that IAM might return fewer than
	// the MaxItems number of results even when there are more results available. We
	// recommend that you check IsTruncated after every call to ensure that you
	// receive all your results.
	IsTruncated [bool](https://pkg.go.dev/builtin#bool)

	// When IsTruncated is true , this element is present and contains the value to use
	// for the Marker parameter in a subsequent pagination request.
	Marker *[string](https://pkg.go.dev/builtin#string)

	// A list containing information about managed policies.
	Policies [][types](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/iam@v1.22.5/types).[ManagedPolicyDetail](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/iam@v1.22.5/types#ManagedPolicyDetail)

	// A list containing information about IAM roles.
	RoleDetailList [][types](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/iam@v1.22.5/types).[RoleDetail](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/iam@v1.22.5/types#RoleDetail)

	// A list containing information about IAM users.
	UserDetailList [][types](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/iam@v1.22.5/types).[UserDetail](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/iam@v1.22.5/types#UserDetail)

	// Metadata pertaining to the operation's result.
	ResultMetadata [middleware](https://pkg.go.dev/github.com/aws/smithy-go/middleware).[Metadata](https://pkg.go.dev/github.com/aws/smithy-go/middleware#Metadata)
	// contains filtered or unexported fields
}
```

Reply:
Sorry, I also haven't understood your consideration. Then I have some questions.

The current code uses [2 filters](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/aws/resources/services/iam/policies.go#L54) `types.EntityTypeAWSManagedPolicy` and `types.EntityTypeLocalManagedPolicy`. When we use this filters, we only get `Policies` as results. So, we only see empty arrays on other fields. If we want to fetch and save all data from this API, I have to remove these filters unlike original codes.

Then, all of these response in `GroupDetailList`, `Policies`, `RoleDatailList` and `UserDatailList` have different schema. In my sight, to save all of them I have to define another tables for each results. But IAM groups, roles and users are already fetched in `aws_iam_groups`, `aws_iam_roles` and `aws_iam_users` tables. So, I can define new of them but is it valuable for CloudQuery to define new tables for duplicated results of `GroupDetailList`, `RoleDatailList` and `UserDatailList`?

My goal on this pull request is simply fetching all IAM policies with `ListPolcies` API instead of `GetAccountAuthorizationDetails` API which fetches only attached policies. The differences are:
- the amount of fetching IAM policies 
- splitting policy_versions onto other table which enables users to choose if they fetch all versions or not. 

I believe this change might be valuable for CloudQuery users. But I understand without committer's agreement my suggestion will not be merged. Sorry for my poor description of this pull request, and I say thank you for your reviewing effort. If you have another questions, feel free to ask me.